pkg/apis/networking/validation/validation.go

```diff
@@ -234,6 +237,7 @@ func ValidateIngressUpdate(ingress, oldIngress *networking.Ingress, requestGV sc
 		// Only require regex path validation for this Ingress if the previous
 		// version of the Ingress also passed that validation.
 		requireRegexPath: allPathsPassRegexValidation(oldIngress),
+		allowResourceBackend: false,
```

Comment:
I think this would need to be changed have similar logic to `requireRegexPath`, but instead of resource backend is specified anywhere in `oldIngress`, this value should be true.